pkg/kv/mem/store.go

```diff
@@ -48,16 +49,19 @@ func encodeKey(key []byte) string {
 	return base64.StdEncoding.EncodeToString(key)
 }
 
-func (s *Store) Get(_ context.Context, partitionKey, key []byte) (*kv.ValueWithPredicate, error) {
+func (s *Store) Get(ctx context.Context, partitionKey, key []byte) (*kv.ValueWithPredicate, error) {
 	if len(partitionKey) == 0 {
 		return nil, kv.ErrMissingPartitionKey
 	}
 	if len(key) == 0 {
 		return nil, kv.ErrMissingKey
 	}
+	if ctx.Err() != nil {
+		return nil, ctx.Err()
+	}
```

Comment:
Just a thought - probably not for this PR - maybe we should refactopr all these checks to a `verifyXXX` function? It can serve other (possibly future) drivers too. WDYT?